backend/backend/Controllers/UsersController.cs

```diff
@@ -57,20 +57,78 @@ public ActionResult<User> GetCurrentUser()
                 return NotFound();
             }
 
-            return user;
+            UserDto userDto = new UserDto()
+            {
+                Id = user.Id,
+                FirstName = user.Firstname,
+                LastName = user.Lastname,
+                Email = user.Email
+            };
+
+            return userDto;
         }
 
         [HttpPost, Route("authentication"), AllowAnonymous]
-        public ActionResult<string> Authenticate([FromBody]UserCredentials userCredentials)
+        public ActionResult<string> Authenticate(string email, string password)
```

Comment:
Oh yeah, I completely missed this, but you should check if the user is confirmed or not. If he's unconfirmed the endpoint should probably send a 403. Now anyone can login after registering no prob, no need to activate the account

Reply:
Sorry for missing this. I believed it was already implemented